src/chunkserver/copyset_node.cpp

```diff
@@ -290,13 +318,31 @@ void CopysetNode::on_shutdown() {
 
 void CopysetNode::on_snapshot_save(::braft::SnapshotWriter *writer,
                                    ::braft::Closure *done) {
+    snapshotFuture_ =
+        std::async(std::launch::async,
+            &CopysetNode::save_snapshot_background, this, writer, done);
+}
+
+void CopysetNode::WaitSnapshotDone() {
+    // wait async snapshot done
+    if (snapshotFuture_.valid()) {
+        snapshotFuture_.wait();
+    }
+}
+
+void CopysetNode::save_snapshot_background(::braft::SnapshotWriter *writer,
```

Comment:
Is there some data lost or some effect If save snapshot background fail?

Reply:
snapshot will fail, and braft will handle it